pkg/osvscanner/osvscanner.go

```diff
@@ -37,6 +38,7 @@ type ScannerActions struct {
 	NoIgnore             bool
 	DockerContainerNames []string
 	ConfigOverridePath   string
+	Exclusions           []string
```

Comment:
I'm softly leaning towards saying we should be taking a slice of regexps here because that is the most accurate (and thus least-error-prone) representation of the data; the reason I'm not completely sold is that I expect a number of library consumers are probably taking user input aka dynamic regexps, but for us it should mean the library doesn't have to handle "what if the string isn't a valid regexp"...

This would be a reason to keep `exclusions` as a public package as then downstream users could just pull that in like the CLI does rather than have to implement their own parser (then again they could also just copy and paste, which is the Go way 🤷)

Reply:
Hey @G-Rath, given in my brain, the user was always going to be input from the user, I naturally just assumed it was going to be a list of string. If it were going to be regexp, we would still need to take the list of string from the flag and parse them as regexp _before_ we assign them to the ScannerActions struct? 🤔

Reply:
What you've said is correct but the code base is made up of both the CLI and the library - and it's the CLI that handles the fact the input is coming in as a string (which is because that's just how data comes in from terminals)

The library on the other hand can just accept regexp so it doesn't need to be responsible for handling the edge cases that come with parsing strings into regexp

Reply:
Thanks @G-Rath that makes sense. Let me shuffle some things and see what happens. 🔀

Reply:
Updated @G-Rath.

The `ScannerActions` struct now has the `Exclusions` field as a `[]*regex.regexp`.